### ISSUE TITLE
reduced "Your full name" string

### ISSUE THREAD

**Issue by HappyPony** [redmatrix/hubzilla#899]
I changed the `Your full name`-Value for https://hub.libranet.de/channel/wallzilla in "Profile Details -> Personal", but I see that in my posts only 91 character as name. **Is it a bug or feature?**

Hubzilla Version 2.8

Maybe important
- The value in _Your full name_ field is 147 character long.
- the missed character are cyrillic character

**Comment by zotlabs** [comment]
I could not duplicate on my site. I vaguely remember that the hub.libranet.de admin was experimenting with utf8mb4 encoding before it was standard in the project and may have DB fields with different lengths than we now use (191 chars, which is the maximum length that mysql can use in an index). The name is actually 180 bytes long (147 characters, of which several characters - the Cyrillic ones - are multibyte UTF-8 sequences).  I've created a test profile with that exact string and posts appear correctly although it is partially truncated using ellipses '...' on small screen widths.

So I believe this is a site configuration issue.

**Comment by AlfredSK** [comment]
@zotlabs Yes, I have switched to utf8mb4/utf8mb4_general_ci before it was officially supported. I think the field we are talking about is 'fullname' in table 'profile'.
The structure of the field looks like this:

```fullname | char(255) | utf8mb4_unicode_ci```

Is this okay?

**Comment by zotlabs** [comment]
It's fine in profile.fullname, but getting truncated when it is copied to xchan.xchan_name, which I think may have a smaller size than 191 on your site.

**Comment by AlfredSK** [comment]
Ah! Okay. Wait a minute... :-)

Hmm. Looks exactly the same.

```xchan_name | char(255) | utf8mb4_unicode_ci```

**Comment by zotlabs** [comment]
Well ya' got me stumped. in Zotlabs/Modules/Profiles.php:L608 we set xchan_name to the same name that we saved to the profile a few lines earlier. There's nothing to mangle it in between. Right after that we set channel.channel_name again to the same thing. You might have a look and see if maybe that one is truncated, but otherwise I can't fix the problem because I can't duplicate it no matter how hard I try.

**Comment by AlfredSK** [comment]
```channel_name | char(255) | utf8mb4_unicode_ci```

Looking good. I can try to reproduce and watch the debug log later this evening. I have an appointment now. :-)

**Comment by zotlabs** [comment]
Current status: UNRESOLVED as we are unable to duplicate or debug further on the affected hub. It is being left open in case somebody else encounters it and/or can shed some light on the issue. So WATCH/WAIT.
